### PR TITLE
Add missing `WindowClient.ancestorOrigins`

### LINKED ISSUE #29281: Missing property `WindowClient.ancestorOrigins`
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/WindowClient

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

Missing property `WindowClient.ancestorOrigins`

### What did you expect to see?

Adding a page for `WindowClient.ancestorOrigins`, and update the description of `WindowClient` interface

### Do you have any supporting links, references, or citations?

https://w3c.github.io/ServiceWorker/#windowclient

image

image

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/windowclient`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/WindowClient
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/windowclient/index.md
* Last commit: https://github.com/mdn/content/commit/acfe8c9f1f4145f77653a2bc64a9744b001358dc
* Document last modified: 2023-07-07T07:19:19.000Z

</details>

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

add missing `WindowClient.ancestorOrigins` documentation

### Additional details

https://w3c.github.io/ServiceWorker/#client-interface

https://w3c.github.io/ServiceWorker/#client-ancestororigins

### Related issues and pull requests

Fixes #29281

### CODE REVIEW

**Review by wbamberg — COMMENTED** [review]
Thank you for adding this, but I think it would be good to expand a bit on what "ancestor origins" means, otherwise this is a bit unhelpful - "`ancestorOrigins` lists ancestor origins" - well, I could have worked that much out from the name, but what does that mean?

AIUI this is the analog in `WindowClient` of https://developer.mozilla.org/en-US/docs/Web/API/Location/ancestorOrigins, right? So could we say something like:

```
The **`ancestorOrigins`** read-only property of the {{domxref("WindowClient")}} interface is an array of strings listing the origins of all ancestors of the browsing context represented by this `WindowClient`.

The first element in the array is the origin of this window's parent, and the last element is the origin of the top-level browsing context. If this window is itself a top-level browsing context, then `ancestorOrigins` is an empty array.
```

Does that sound correct and helpful?

**Comment by skyclouds2001** [comment]
> Thank you for adding this, but I think it would be good to expand a bit on what "ancestor origins" means, otherwise this is a bit unhelpful - "`ancestorOrigins` lists ancestor origins" - well, I could have worked that much out from the name, but what does that mean?
> 
> AIUI this is the analog in `WindowClient` of https://developer.mozilla.org/en-US/docs/Web/API/Location/ancestorOrigins, right? So could we say something like:
> 
> ```
> The **`ancestorOrigins`** read-only property of the {{domxref("WindowClient")}} interface is an array of strings listing the origins of all ancestors of the browsing context represented by this `WindowClient`.
> 
> The first element in the array is the origin of this window's parent, and the last element is the origin of the top-level browsing context. If this window is itself a top-level browsing context, then `ancestorOrigins` is an empty array.
> ```
> 
> Does that sound correct and helpful?

Thanks for you suggestion!

File: files/en-us/web/api/windowclient/ancestororigins/index.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+{{APIRef("Service Workers API")}}
+
+The **`ancestorOrigins`** read-only property of the {{domxref("WindowClient")}} interface is an array of strings listing the origins of all ancestors of the browsing context represented by this `WindowClient` in reverse order.
+
+The first element in the array is the origin of this window's parent, and the last element is the origin of the top-level browsing context. If this window is itself a top-level browsing context, then `ancestorOrigins` is an empty array.
+
+## Value
+
+An array of string.
```
Comment (wbamberg) [inline files/en-us/web/api/windowclient/ancestororigins/index.md:17]:
```suggestion
An array of strings.
```

File: files/en-us/web/api/windowclient/index.md [hunk 2]
```diff
@@ -24,6 +24,8 @@ _`WindowClient` inherits methods from its parent interface, {{domxref("Client")}
 
 _`WindowClient` inherits properties from its parent interface, {{domxref("Client")}}._
 
+- {{domxref("WindowClient.ancestorOrigins")}} {{ReadOnlyInline}}
+  - : An array of string that indicates the ancestor origins of the browsing context represented by this `WindowClient` in reverse order.
```
Comment (wbamberg) [inline files/en-us/web/api/windowclient/index.md:28]:
```suggestion
  - : An array of strings that indicates the ancestor origins of the browsing context represented by this `WindowClient` in reverse order.
```

**Review by wbamberg — APPROVED** [review]
👍 thank you @skyclouds2001 !

**Pull request merged**